include/mesh/checkpoint_io.h

```diff
@@ -99,16 +99,44 @@ class CheckpointIO : public MeshInput<MeshBase>,
   bool   binary() const { return _binary; }
   bool & binary()       { return _binary; }
 
+  /**
+   * Get/Set the flag indicating if we should read/write binary.
+   */
+  bool   parallel() const { return _parallel; }
+  bool & parallel()       { return _parallel; }
 
   /**
    * Get/Set the version string.
    */
   const std::string & version () const { return _version; }
   std::string &       version ()       { return _version; }
 
+  /**
+   * Get/Set the processor_id to use
+   */
+  const processor_id_type & current_processor_id() const { return _my_processor_id; }
+  processor_id_type & current_processor_id() { return _my_processor_id; }
+
+  /**
+   * Get/Set the n_processors to use
+   */
+  const processor_id_type & current_n_processors() const { return _my_n_processors; }
+  processor_id_type & current_n_processors() { return _my_n_processors; }
+
```

Comment:
Actually - those are for `m->n` splitting.  Like run with `m` MPI processes... but create output files for `n` MPI processes.

To do that a code can loop over and set `current_n_processors()` and `current_processor_id()` and call `write()`... and the file that comes out will be for _that_ situation.

I'll update the comment